sympy/testing/runtests.py

```diff
@@ -729,6 +729,18 @@ def _get_doctest_blacklist():
             "sympy/parsing/sym_expr.py",
         ])
 
+    if import_module("scipy") is None:
+        # throws ModuleNotFoundError when scipy not installed
+        blacklist.extend(
+            ["doc/src/guides/solving/solve-ode.md",]
+        )
+
+    if import_module("numpy") is None:
+        # throws ModuleNotFoundError when scipy not installed
+        blacklist.extend(
+            ["doc/src/guides/solving/solve-ode.md",]
+        )
+
```

Comment:
A doctest for solve-ode.md uses both SciPy and NumPy, so the doctest fails if those optional dependencies aren't installed. I thought it made more sense to add separate `if import_module("<module_name>") is None:` blocks for each (SciPy and NumPu) and add solve-ode.md to the blacklist in each block because
- there's another page which ([solve-numerically.md](https://github.com/sympy/sympy/pull/23853/files#diff-22796efbef97cd6b88b4d9cab6f86a61131f535414899b9f90d74baa6c1a3ce3) from #23853) which is added to the blacklist if scipy isn't installed, so 
- it seems more logical to me to have a block for each dependency rather than for each page.

I don't expect it will cause a problem for a page (solve-ode.md) to be added to the blacklist twice; please let me know if it would. An alternative is to have a single if block which added solve-ode.md to the blacklist if either dependency isn't installed.